src/goSuggest.ts

```diff
@@ -235,6 +230,9 @@ export class GoCompletionItemProvider implements vscode.CompletionItemProvider {
 			});
 		});
 
+		return Promise.all([setPkgsList, setGocodeProps]).then(() => {
```

Comment:
With the `-format` option I am guessing that `gopkgs` is a bit slower than when you made the comparison between `go list all` and `gopkgs` in the description of this PR.

What are your current numbers?

Every new word typed triggers completion. So you can imagine that `gopkgs` will be run for every word typed. 
Some users have a whole lot of packages that `go list all` took over 10 seconds to finish. How does `gopkgs -format` fare?

I still believe we should use cache and have the cache updated at a regular interval instead of running `gopkgs` for every word.

What do you think?

Reply:
`gpkgs -format` faster than `go list all`

For 3382 packages packages:
```
go list all  4.62s user 3.27s system 124% cpu 6.360 total
gopkgs  0.65s user 0.70s system 264% cpu 0.508 total
```
6.36s vs 0.5s, big difference.
This gopkgs using fastwalk, faster than golang standard `fs.Walk()`.

Not every word typed, it only invoked on only when completion triggered. Typing "strings" 7 words typed will only cost single `gopkgs` invocation. So this is quite cheap and fast enough.

Cache will be much better of course.
If we cache for 10 or 20 second, user probably can create new file in new folder with new package faster than cache expiry time. So new package will not provided by completion. The workaround crossed in my mind is we need to some kind of file system watcher and check if there is changed file, then invalidate the cache or run `gopkgs` immediately to refresh the cache, etc. My suggestion is we can do this cache things on another PR.

Reply:
Changed my mind... let's try to use 5s cache expiry

Reply:
> Not every word typed, it only invoked on only when completion triggered.

Completion is triggered for every word typed. I think you misunderstood my statement as every "letter" typed

Reply:
Ahh... yup..